### PR TITLE
[#1342] Unification of modal templates

### LINKED ISSUE #1342: Change modal loading
Now we have 3 modals in the mp:
- modal for creating projects by service ordering
- modal for question in services view
- modal for issue report to the xGUS

All of these use different templates for generating modals, and we have generic wizard for controllers which uses one template.

There is a need to refactor all these modals to make every dependent of this one modal template `view/layouts/_modal.html.haml`

### PR DESCRIPTION
**Pull request by goreck888**
Change the way for rendering modals in mp.
Now for all modals we use one template
Fixes #1342

### CODE REVIEW

File: app/controllers/services/questions_controller.rb [hunk 1]
```diff
@@ -33,12 +34,13 @@ def create
...
     def render_modal_form
-      render "services/question_modal",
+      render "layouts/show_modal",
              content_type: "text/javascript",
              locals: {
                 title: "Ask provider",
                 action_btn: t("simple_form.labels.question.new"),
-                form: "services/questions/form"
+                form: "services/questions/form",
+                variables: { service: @service, question: @question }
```
Comment (mkasztelnik) [inline app/controllers/services/questions_controller.rb]: Maybe `form_locals` is a better name?

File: app/controllers/reports_controller.rb [hunk 2]
```diff
@@ -1,6 +1,15 @@
 # frozen_string_literal: true
 
 class ReportsController < ApplicationController
+  def new
+    @report = Report.new
+
+    respond_to do |format|
+      format.html
```
Comment (mkasztelnik) [inline app/controllers/reports_controller.rb]: Is `html` needed here?

File: app/controllers/services/questions_controller.rb [hunk 3]
```diff
@@ -3,6 +3,7 @@
 class Services::QuestionsController < ApplicationController
   def new
     @question = Service::Question.new
+    @service = Service.friendly.find(params[:service_id])
 
     respond_to do |format|
       format.html
```
Comment (mkasztelnik) [inline app/controllers/services/questions_controller.rb]: Is there a situation when question form is rendered as html?

File: app/views/reports/new.html.haml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+.container
+  .mb-4.pb-4.border-header
+    %h1.mr-auto.my-auto Ask provider
+
+  = render "reports/form", remote: true, report: @report
```
Comment (mkasztelnik) [inline app/views/reports/new.html.haml]: It looks like the `remote` variable is not used in the `reports/form` partial.

File: app/views/reports/new.html.haml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+.container
```
Comment (mkasztelnik) [inline app/views/reports/new.html.haml]: It this view used anywhere in the application? I do not reckon a place in the marketplace were report is rendered as normal html (only js).

File: app/views/services/questions/new.html.haml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+.container
+  .mb-4.pb-4.border-header
+    %h1.mr-auto.my-auto Ask provider
+
+  = render "services/questions/form", remote: true, question: @question, service: @service
```
Comment (mkasztelnik) [inline app/views/services/questions/new.html.haml]: `remote` variable is not used in `services/questions/form`

File: app/views/services/questions/new.html.haml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+.container
```
Comment (mkasztelnik) [inline app/views/services/questions/new.html.haml]: Similar to the above: I do not reckon a place in MP where the question form is rendered as html (only js).

File: app/controllers/reports_controller.rb [hunk 8]
```diff
@@ -1,6 +1,15 @@
...
+      format.js { render_modal_form }
+    end
+  end
+
   def create
```
Comment (mkasztelnik) [inline app/controllers/reports_controller.rb]: If report is created only using `js` this method can also be cleaned a bit and all `html` elements can be removed.

File: app/views/projects/_form.html.haml [hunk 9]
```diff
@@ -1,4 +1,4 @@
-= simple_form_for @project, remote: @show_as_modal,
+= simple_form_for project, remote: show_as_modal,
```
Comment (mkasztelnik) [inline app/views/projects/_form.html.haml]: I propose to introduce one small change here:

```ruby
#change 
remote: show_as_modal
#to 
local_assigns[:show_as_modal]
```

This small change will allow us to have default `show_as_modal` value set to `false`. As a conclusion you will be able to remove it from the normal `projects/{new/edit}.htm.haml` partials.

File: app/controllers/reports_controller.rb (hunk 8 above)
Comment (mkasztelnik) [reply app/controllers/reports_controller.rb]: This is also valid for questions controller.

File: app/controllers/services/questions_controller.rb (hunk 1 above)
Comment (goreck888) [reply app/controllers/services/questions_controller.rb]: No problem, I just saved previous naming :)

File: app/controllers/services/questions_controller.rb (hunk 3 above)
Comment (goreck888) [reply app/controllers/services/questions_controller.rb]: No, I delete it.

File: app/views/reports/new.html.haml (hunk 5 above)
Comment (goreck888) [reply app/views/reports/new.html.haml]: No it isn't used, I delete this file.

File: app/views/services/questions/new.html.haml (hunk 6 above)
Comment (goreck888) [reply app/views/services/questions/new.html.haml]: File deleted.

File: app/views/services/questions/new.html.haml (hunk 7 above)
Comment (goreck888) [reply app/views/services/questions/new.html.haml]: File deleted.

File: app/controllers/reports_controller.rb (hunk 8 above)
Comment (goreck888) [reply app/controllers/reports_controller.rb]: Done.

File: app/views/projects/_form.html.haml (hunk 9 above)
Comment (goreck888) [reply app/views/projects/_form.html.haml]: Done.

File: app/views/reports/new.html.haml (hunk 4 above)
Comment (goreck888) [reply app/views/reports/new.html.haml]: File deleted.

File: app/controllers/reports_controller.rb (hunk 2 above)
Comment (goreck888) [reply app/controllers/reports_controller.rb]: No, I delete it.

**Comment by agpul** [comment]
I have a question. Is it intentional that on production the template looks nice (graphic, fonts etc.) and in this PR we've lost this nice view?
Production: 
image
PR: 
image

**Comment by mkasztelnik** [comment]
Nope, this is a bug. Most probably CSS was created to specific and should be improved. @goreck888 can you take a look at this issue?

**Pull request merged**